saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/MultiModalSpRLDataModel.scala

```diff
@@ -288,6 +288,29 @@ object MultiModalSpRLDataModel extends DataModel {
       pos(first) + "::" + pos(second)
   }
 
+  val imageConfirmsRelation = property(pairs, cache = true) {
+    var result = false
+    r: Relation =>
+      val (x, rel) = getArguments(r)
+      val img = ((pairs(r)<~sentenceToRelations<~documentToSentence)~>documentToImage).head
+      val seg = images(img)~> imageToSegment
+
+      val imageRelations = segmentRelations.getAllInstances.filter(sr=> sr.getImageId().equals(img.getId()))
+      imageRelations.foreach(s => {
+        val firstConcept = getSegmentConcepts(img.getId(), s.getFirstSegmentId())
+        val secondConcept = getSegmentConcepts(img.getId(), s.getSecondSegmentId())
+        val relation = s.getRelation()
+        if ((x.getText().toLowerCase().equals(firstConcept.toLowerCase()) || x.getText().toLowerCase().equals(secondConcept.toLowerCase())) && (rel.getText().toLowerCase().equals(relation.toLowerCase()))) {
```

Comment:
I think it will be very very small possibility that the image concept equals the tr or lm text, so this will not work.
The second thing is there are multiple concepts per segment, you want to compare the text with which one?